x-pack/metricbeat/module/gcp/metrics/metricset.go

```diff
@@ -228,7 +247,7 @@ func (m *MetricSet) metricDescriptor(ctx context.Context, client *monitoring.Met
 
 	for _, sdc := range m.MetricsConfig {
 		for _, mt := range sdc.MetricTypes {
-			req.Filter = fmt.Sprintf(`metric.type = starts_with("%s")`, sdc.ServiceName+".googleapis.com/"+mt)
+			req.Filter = fmt.Sprintf(`metric.type = starts_with("%s%s")`, sdc.MetricPrefix(), mt)
```

Comment:
These names seem to have URL semantics, you can likely rely on `path.Join` and forget about caring if the user put or not a `/` after the setting :slightly_smiling_face: 
```suggestion
			req.Filter = fmt.Sprintf(`metric.type = starts_with("%s")`, path.Join(sdc.MetricPrefix(), mt))
```